Procedure script implementation

christopher-gong opened the pull request:
I've implemented the procedure script in the code directory. It can now take a nml file as input, and output a csv file. with the following information: ['DATE_EST', 'STORY_ID', 'TICKER', 'CLOSEST_ID', 'CLOSEST_SCORE', 'TOTAL_OVERLAP', 'IS_OLD', 'IS_REPRINT', 'IS_RECOMB']. Please take a look in newprocedure.py.

christopher-gong committed:
everything from story

christopher-gong committed:
begun replication techniques

christopher-gong committed:
replication progress

christopher-gong committed:
written procedure

hodsonjames reviewed (commented):
We should clean this up so that you commit one version of the 'procedure'. Also, we need better names than 'procedure/newProcedure'--think of something descriptive :) Looks like we still need to split this into modules to make it easy to extend with new measures, and please don't forget to allow the script to take a directory of nml files that it should read in alphabetical order.

christopher-gong committed:
procedure update with additional params

christopher-gong committed:
modified from list to plain text

christopher-gong committed:
write straight to file

christopher-gong committed:
large code refactoring, organization, documentation, and other improvements

christopher-gong commented:
I've implemented the major changes below:
1. Split the file into replication and measures
2. Writing directly to CSV file
3. Order nml's numerically
4. Renamed Procedure
5. Comment and Refactor code

I've also implemented several other minor changes for code improvement. Please take a look in staleNewsProcedure.py.

christopher-gong committed:
main method changes

christopher-gong commented:
I've made the minor change of adding the main method, along with the changes in the previous comment. Thank you!
